### PR TITLE
Adds Validate=false that disables validation when applying resources.…

### LINKED ISSUE #1063: Add ability to kubectl apply objects using `--validate=false`
For a certain class of group kinds, it is necessary to `kubectl apply` them using the `--validate=false` flag. Examples of this are kubernetes types which uses `RawExtension`, such as [ServiceCatalog](https://github.com/kubernetes-incubator/service-catalog/blob/master/pkg/apis/servicecatalog/v1beta1/types.go#L497).

This feature request is to allow a user to disable validation of an object when running `kubectl apply` for an entire group/kind. For example, all users would naturally want to disable validation for `servicecatalog.k8s.io/ServiceCatalog`. 

This feature request is closely related to the new feature for custom health assessments, in that the configuration option for disabling validation would be a peer option to the health lua script.

### PR DESCRIPTION
**Pull request by alexec**
… Closes #1063, #1734, #1751

### CODE REVIEW

File: controller/sync.go [hunk 1]
```diff
@@ -475,7 +475,8 @@ func (sc *syncContext) setOperationPhase(phase v1alpha1.OperationPhase, message
 
 // applyObject performs a `kubectl apply` of a single resource
 func (sc *syncContext) applyObject(targetObj *unstructured.Unstructured, dryRun bool, force bool) (v1alpha1.ResultCode, string) {
-	message, err := sc.kubectl.ApplyResource(sc.config, targetObj, targetObj.GetNamespace(), dryRun, force)
+	validate := !resource.HasAnnotationOption(targetObj, common.AnnotationSyncOptions, "LastValidate=false")
```
Comment (alexec) [inline controller/sync.go]: The key code change.

File: test/e2e/helm_test.go [hunk 2]
```diff
@@ -41,3 +41,40 @@ func TestHelmCrdInstallIsCreated(t *testing.T) {
 			return p.Name == "hook"
 		}))
 }
+
+func TestServiceCatalog(t *testing.T) {
+	t.Skip("slow")
```
Comment (alexec) [inline test/e2e/helm_test.go]: we skip these tests - need to figure out the correct e2e test for this - @jessesuen any suggestions?

File: util/kube/ctl.go [hunk 3]
```diff
@@ -227,6 +227,9 @@ func (k KubectlCmd) ApplyResource(config *rest.Config, obj *unstructured.Unstruc
 	if force {
 		applyArgs = append(applyArgs, "--force")
 	}
+	if !validate {
+		applyArgs = append(applyArgs, "--validate=false")
```
Comment (alexec) [inline util/kube/ctl.go]: Another key change.

File: util/kube/kubetest/mock.go [hunk 4]
```diff
@@ -19,29 +19,31 @@ type MockKubectlCmd struct {
 	APIResources []kube.APIResourceInfo
 	Commands     map[string]KubectlOutput
 	Events       chan watch.Event
+	LastValidate bool
```
Comment (alexec) [inline util/kube/kubetest/mock.go]: Guess I could have made this `*bool` so we could change it, but that seemed odd.

File: util/settings/settings.go [hunk 5]
```diff
@@ -539,6 +539,15 @@ func (mgr *SettingsManager) SaveSettings(settings *ArgoCDSettings) error {
 	} else {
 		delete(argoCDCM.Data, repositoryCredentialsKey)
 	}
+	if len(settings.HelmRepositories) > 0 {
```
Comment (alexec) [inline util/settings/settings.go]: Enables hot-reload for Helm repos creds.

**Comment by alexec** [comment]
The code coverage is complaining about files that have not changed in this PR.

**Review by jessesuen — APPROVED** [review]

**Pull request merged**